### ISSUE TITLE
[0.50.02-SNAPSHOT] Round Report text is sequenced incorrectly

### ISSUE THREAD

**Issue by PhoenixHeart512** [MegaMek/megamek#6299]
### Prerequisites and Pre-Issue Checklist

- [X] I'm reporting the issue to the correct repository:
- [MegaMek](https://github.com/MegaMek/megamek/issues)
- [MegaMekLab](https://github.com/MegaMek/megameklab/issues)
- [MekHQ](https://github.com/MegaMek/mekhq/issues)

- [X] I've tested the issue against at least the latest MILESTONE version
- [X] I've asked on the MegaMek Discord about the error
- [X] I've reviewed the BattleTech rules and MegaMek documentation, and I've confirmed that something isn't working as intended.
- [X] I've searched the Github tracker and haven't found the issue listed

### Severity *

Low (Minor/Nuisance): Minor glitches or cosmetic issues that don’t affect gameplay and occur rarely.

### Brief Description *

The round report where this Viking skidded into a building (and maybe also the Grim Reaper) has their rolls and damage mixed together and alternating.

Not sure if this is intended behavior or not due to involving a skid and collision, but it makes the report really hard to follow.

![image](https://github.com/user-attachments/assets/9c13cb26-ceab-4fb7-9872-420fe833fe99)

I've included the save file from when the report was open, as well as the two previous autosaves to help with replication if needed.

### Steps to Reproduce

Possibly only occurs when involving a skid/collision between two units in an urban environment?

### Operating System *

Windows 10

### Java Version *

Java Version: 17.0.6

### MegaMek Suite Version *

Free Text (type manually)

### Custom MegaMek Version

megamek.MegaMek.initializeLogging(MegaMek.java:146) - Starting MegaMek v0.50.02-SNAPSHOT     Build Date: 2024-12-20T03:22:42.532160220     Today: 2024-12-20     Origin Project: MekHQ     Java Vendor: Eclipse Adoptium     Java Version: 17.0.6     Platform: Windows 10 10.0 (amd64)     System Locale: en_US     Total memory available to MegaMek: 8 GB     MM Code Revision: 8ad58dcbbf41e73ffd3e20b6a9bb3d159d6616dd     MML Code Revision: 543ecc92823ad4c939d7f5316c0075cd50616d25     MHQ Code Revision: f1a5c049c57b3a692995ae0039836f34b46fe8bd

### Attach Files

[RoundReportBug.zip](https://github.com/user-attachments/files/18213727/RoundReportBug.zip)

### Final Checklist

- [X] I've checked to make sure that this issue has not already been filed
- [X] I'm reporting only one issue in this ticket for clarity and focus

**Comment by Sleet01** [comment]
A very smart and productive tester had also previously reported a similar skid + building + charge issue:
https://github.com/MegaMek/megamek/issues/3837

Probably at least somewhat connected.

**Comment by Sleet01** [comment]
This is actually the same issue as #6434 and #3890 : old code that writes directly into the Phase Report ahead of the report for the movement itself.  In this case, note that the Viking failed a skill check (correctly reported at the top), then skidded into Grim Reaper in Hex 2116.  The interleaved (and apparently out-of-order) reporting is caused by _some_ of these checks writing into a temporary Report Vector, but the others writing directly into the vPhaseReport for this phase.

I will apply the same fix used for the two issues above.
